fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -367,15 +405,16 @@ public void syncMetaData() throws IOException {
     public void rename(@Nonnull final String source, @Nonnull final String dest) {
         LOGGER.trace("rename -> source={}, dest={}", source, dest);
         final byte[] key = metaSubspace.pack(source);
-        CompletableFuture<byte[]> completableFuture;
-        completableFuture = context.ensureActive().get(key).exceptionally( (e) -> {
-            throw new RecordCoreArgumentException("Invalid source name in rename function for source: " + source);
-        });
-        completableFuture.thenAcceptAsync( (value) -> {
-            this.fileReferenceCache.invalidate(source);
+        context.asyncToSync(FDBStoreTimer.Waits.LUCENE_RENAME, context.ensureActive().get(key).thenApply((Function<byte[], Void>)value -> {
+            if (value == null) {
+                throw new RecordCoreArgumentException("Invalid source name in rename function for source: " + source);
```

Comment:
Feel free to ignore this for now, but we have a few general rules for our exceptions:

* The text of the exception should be static (no variables in it)
* Logging properties should be added to give context to the error
* The logging properties should provide as much detail as possible to diagnose the source of the error

The idea here is that by having the text of the exception static, that it makes it trivial to search through all logs in the system and find all instances of the error, the extra context information (in this case "source"), can then be used for grouping/filtering/etc. without any regex games when searching. 

Also, when you are testing, it may be trivial to understand the context of this message (maybe it is a single record store with only one Lucene index), but as things scale, and you get millions of databases and maybe even some with multiple lucene indexes, having context on at least which index it is can be helpful.

This is all a long winded way of saying, probably this should be:

```
throw new RecordCoreArgumentException("Invalid source name in rename function for source")
    .addLogInfo(LogMessageKeys.SOURCE,  source);
```

and, ideally, you probably want additional `LogMessageKey`s that tell you which index, which key, and any other information that would be useful for understanding the context of the issue.

I really need add this info into the contributors guide :)

Reply:
I attempted to address this (in a rough-and-ready sort of fashion). Lemme know how well I did :)